### ISSUE TITLE
Title: Implementing Plant Disease Prediction using ResNet Model

### ISSUE THREAD

**Issue by PRIYANSHU2026** [Akshat111111/Hedging-of-Financial-Derivatives#198]
**Description:**
I am working on implementing a plant disease prediction system using the ResNet model. I plan to use the HAGGLE dataset for training the model. The goal is to train the model on a cloud platform to leverage its computing power and scalability.

**Tasks:**
- [x] Download the KAGGLE dataset.
- [x] Preprocess the dataset for training.
- [x] Implement the ResNet model architecture.
- [x] Train the model on a cloud platform.
- [x] Evaluate the model's performance.
- [ ] Deploy the trained model for inference.

**Additional Features:**
- [x] Import necessary libraries:
  - `import os` for working with files
  - `import numpy as np` for numerical computations
  - `import pandas as pd` for working with dataframes
  - `import torch` for PyTorch module
  - `import matplotlib.pyplot as plt` for plotting information on graphs and images using tensors
  - `import torch.nn as nn` for creating neural networks
  - `from torch.utils.data import DataLoader` for dataloaders
  - `from PIL import Image` for checking images
  - `import torch.nn.functional as F` for functions for calculating loss
  - `import torchvision.transforms as transforms` for transforming images into tensors
  - `from torchvision.utils import make_grid` for data checking
  - `from torchvision.datasets import ImageFolder` for working with classes and images
  - `from torchsummary import summary` for getting the summary of our model

**Labels:** machine learning, cloud computing, enhancement, big project

**Note:** This will be the biggest project in this repository.

**Comment by Akshat111111** [comment]
The repo is of finance not, machine learning

**Comment by Amarta113** [comment]
@Akshat111111 is it necessary to work on finance related tasks?

**Comment by Akshat111111** [comment]
yes, this repo is directly or indirectly related to finance.
